src/gt4py/next/program_processors/runners/dace/transformations/map_fusion_extended.py

```diff
@@ -459,6 +460,27 @@ def can_be_applied(
 
         # TODO(phimuell): Implement a check that the array is single use data?
 
+        # Check that the data read by the nested map are not accessed by any other outgoing edge
+        # of the access node.
+        access_node_out_edges = graph.out_edges(self.access_node)
+        access_node_out_edges_data = [edge.data.data for edge in access_node_out_edges]
+        second_map_in_edges = graph.in_edges(self.second_map_entry)
+        matching_data_subsets = {
+            second_map_edge.data.data: second_map_edge.data.subset
+            for second_map_edge in second_map_in_edges
+            if second_map_edge.data.data in access_node_out_edges_data
```

Comment:
```suggestion
            if  (not second_map_edge.data.is_empty()) and second_map_edge.data.data in access_node_out_edges_data
```
Empty edges are used to enforce a certain ordering, but they do not transfer any data, this is just a precaution.

As a side note.
Above, when you collected `access_node_out_edges_data`, I mentioned that `edge.data.data` in general does not work for `AC -> AC` connections.
However, in this specific case it is valid, because for syntactical reasons all upstream adjacent nodes to a MapEntry node have to be AccessNodes.
So you could keep it that way.
However, I am not sure if changing it to `edge.src.data` might be more readable, but for me both are fine.